server/src/test/java/org/elasticsearch/transport/RemoteConnectionStrategyTests.java

```diff
@@ -75,6 +75,32 @@ public void testChangeInConnectionProfileMeansTheStrategyMustBeRebuilt() {
         assertTrue(first.shouldRebuildConnection(newBuilder.build()));
     }
 
+    public void testCompressionSchemeDefaults() {
+        // Test explicit
+        Settings.Builder explicitBuilder = Settings.builder();
+        explicitBuilder.put(RemoteConnectionStrategy.REMOTE_CONNECTION_MODE.getConcreteSettingForNamespace("cluster-alias").getKey(),
+            "proxy");
+        explicitBuilder.put(ProxyConnectionStrategy.PROXY_ADDRESS.getConcreteSettingForNamespace("cluster-alias").getKey(),
+            "127.0.0.1:9300");
+        explicitBuilder.put(RemoteClusterService.REMOTE_CLUSTER_COMPRESS.getConcreteSettingForNamespace("cluster-alias").getKey(),
+            randomFrom("true", "indexing_data", "false"));
+        explicitBuilder.put(TransportSettings.TRANSPORT_COMPRESSION_SCHEME.getKey(), "lz4");
+        ConnectionProfile connectionProfile = FakeConnectionStrategy.buildConnectionProfile("cluster-alias", explicitBuilder.build());
+        assertEquals(Compression.Scheme.DEFLATE, connectionProfile.getCompressionScheme());
```

Comment:
We may have it elsewhere, but if not, could we add assertions to validate that if `REMOTE_CLUSTER_COMPRESSION_SCHEME` is set, we use the value set there?